lib/Marlin/Marlin/src/feature/bed_preheat.cpp

```diff
@@ -38,11 +38,12 @@ void BedPreheat::update() {
 }
 
 uint32_t BedPreheat::required_preheat_time() {
-    if (thermalManager.degTargetBed() < minimal_preheat_temp) {
+    const uint32_t target_temp = uint32_t(thermalManager.degTargetBed());
+    if (target_temp < uint32_t(minimal_preheat_temp)) {
         return 0;
     }
 
-    return std::max((180 + (thermalManager.degTargetBed() - 60) * (12 * 60 / 50)) * 1000, 0);
+    return std::max<uint32_t>((180 + (target_temp - 60) * (12 * 60 / 50)) * 1000, int32_t(0));
```

Comment:
This will not work properly, `std::max<uint32_t>` will cast the values to uint32_t before maxing, so the left side would always be >= 0.

Reply:
Fixed that issue.